### PR TITLE
Create room modal: functionality

### PR DESCRIPTION
**Pull request by bearni95**
- Added functionality to manually modify a room's name upon creation
- Reworked interaction with lib `unique-names-generator` to use straight-away as room names
- Fully working visual modal to create a room with a pre-generated room name, and the option to manually alter it

### CODE REVIEW

File: src/hooks/useCreateRoom.ts [hunk 1]
```diff
@@ -13,40 +11,45 @@ export default function useCreateRoom(): () => Promise<void> {
...
+            if (!account) {
+                throw new Error('Missing account')
+            }
+
+            if (!metamaskStreamrClient) {
+                throw new Error('Missing metamask streamr client')
+            }
+
+            const stream = await metamaskStreamrClient.createStream({
+                id: `/streamr-chat/room/${roomName}`.toLowerCase(),
```
Comment (mondoreale) [inline src/hooks/useCreateRoom.ts:33]: What if my room name has a white space in it? Gets url-encoded? Or what if the name I'm using already exists?

File: src/components/pages/Chat/AddRoomItem.tsx [hunk 2]
```diff
@@ -179,21 +188,13 @@ function UnstyledAddRoomItem(props: Props) {
                         </ModalHeader>
                         <div>
                             <Subheading>Name</Subheading>
-                            <input
-                                disabled={true}
-                                placeholder="random-name-abc"
-                            />
+                            <input defaultValue={roomName} />
```
Comment (mondoreale) [inline src/components/pages/Chat/AddRoomItem.tsx:191]: This does nothing. `value` and `onChange` are missing thus rooms get created with random names.

Comment (mondoreale) [reply src/components/pages/Chat/AddRoomItem.tsx:191]: `/new roomname` created a room named "roomname", I confirm that.

Comment (bearni95) [reply src/components/pages/Chat/AddRoomItem.tsx:191]: You're totally right, just added the `onChange` handler to update `setRoomName` and it's working alright now

File: src/hooks/useCreateRoom.ts (hunk 1 above)
Comment (bearni95) [reply src/hooks/useCreateRoom.ts:33]: The client rejects most special characters `!?,.+/&`, only allowing `-` and `_`. Whitespaces are also rejected.

Should we warn about this ("Please use alphanumeric characters, `-` and `_` only. The room id will get lowercased automatically") in the text below the romm name on the creation modal?

**Review by mondoreale — APPROVED** [review]

**Pull request merged**